### ISSUE TITLE
pandas resample weekly and interpolate - wrong results

### ISSUE THREAD

**Issue by denfromufa** [pandas-dev/pandas#16381]
https://gist.github.com/denfromufa/fd73a844f474847547eeb684572f2117

**Comment by jreback** [comment]
pls replace the gist with a copy-pastable example and ``pd.show_versions()`` as indicated in the issue request page.

**Comment by denfromufa** [comment]
@jreback done!

**Comment by jreback** [comment]
can you remove the rendered frames. simply run this in ipython and paste the results.

**Comment by denfromufa** [comment]
@jreback what is wrong with frames? i don't work with pandas in ipython terminal.

**Comment by TomAugspurger** [comment]
@denfromufa can you post your expected output?

I think you're getting tripped up by the endpoints. When you do `pdtest.resample('W-MON')`, the array is upsampled

```python
In [42]: pdtest.resample("W-MON")._upsample(None).head()
Out[42]:
             0
1900-02-05 NaN
1900-02-12 NaN
1900-02-19 NaN
1900-02-26 NaN
1900-03-05 NaN
```

Since the original left-endpoint doesn't align with a W-MON freq, you get NaN and then `.interpolate` will be NaN for everything before the first valid (upsampled) observation.

If you resample at a freq that does align with your original first point, it will be I think what you expect:

```python
In [43]: pdtest.resample("W-WED")._upsample(None).head()
Out[43]:
               0
1900-01-31  12.0
1900-02-07   NaN
1900-02-14   NaN
1900-02-21   NaN
1900-02-28  11.0

In [45]: pdtest.resample("W-WED")._upsample(None).interpolate().head()
Out[45]:
                0
1900-01-31  12.00
1900-02-07  11.75
1900-02-14  11.50
1900-02-21  11.25
1900-02-28  11.00
```

**Comment by denfromufa** [comment]
@TomAugspurger this is good explanation, but I expected interpolation even for mis-aligned data. I think for weekly interpolation the safest option to use is like this:

```
pdtest.resample('D').interpolate()[::7]
``` 

But the most and only upvoted answer on SO suggests what I did:
http://stackoverflow.com/a/14531149/2230844

Anyway I'm having even a bigger problem with original weekly interpolation method, let me open another issue for it.

**Comment by TomAugspurger** [comment]
Agreed that it's a surprising output, unless your familiar with how it's implemented. I'm not sure there's much we can do though... Potentially we could fill the endpoints of the upsampled DataFrame with the original endpoints?

```python
# would have to handle dataframe's properly, but this is the main idea
In [34]: up.squeeze().fillna({up.index[0]: pdtest.iloc[0, 0]}).interpolate().head()
Out[34]:
1900-02-04    12.000000
1900-02-11    11.764706
1900-02-18    11.529412
1900-02-25    11.294118
1900-03-04    11.058824
Freq: W-SUN, Name: 0, dtype: float64

In [35]: w = pdtest.resample("W")

In [36]: up = w._upsample(None)

In [37]: up.squeeze().fillna({up.index[0]: pdtest.iloc[0, 0]}).interpolate().head()
Out[37]:
1900-02-04    12.000000
1900-02-11    11.764706
1900-02-18    11.529412
1900-02-25    11.294118
1900-03-04    11.058824
Freq: W-SUN, Name: 0, dtype: float64
```

we would want to look at if that breaks in violations upsampling.

**Comment by denfromufa** [comment]
@TomAugspurger i think one problem here is that syntax with keyword for `fill_method='interpolate'` is deprecated. But the resampling can be dependent on the method like in this case.

**Comment by denfromufa** [comment]
related? 

https://github.com/pandas-dev/pandas/issues/14297

**Comment by eromoe** [comment]
I found this problem too...  when it can be fixed?

**Comment by denfromufa** [comment]
@eromoe here is a workaround:

https://stackoverflow.com/a/44053092/2230844
